compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/window/Dialog.desktop.kt

```diff
@@ -184,7 +189,7 @@ fun Dialog(
                 appliedState.size = state.size
             }
             if (state.position != appliedState.position) {
-                dialog.setPositionSafely(state.position)
+                dialog.setPositionSafely(state.position, WindowLocationTracker.getCascadeLocationFor(dialog))
```

Comment:
Is it expected that Dialog shows up in the middle of a screen/display regardless of the window position?   Is it a requirement for Dialogs? 

![Screenshot 2022-10-04 at 10 29 20](https://user-images.githubusercontent.com/7372778/193772411-da46599e-1f50-4aff-a3e3-601652b1125c.png)

Reply:
I guess, we can change the default position of the dialog to align it in the center of the parent window. Swing alert dialogs (with Yes/No/Cancel buttons) behave this way.

I will try to implement it in next few hours.

Reply:
I realized, that it is better to introduce new API for that: `position = WindowPosition.OwnerAligned(Center)` or something like that. Not aligning it in the center when we use `WindowPosition.PlatformDefault`.

Let's implement it in the future, not now. It is a separate feature. We aligned dialog in the center of the screen from 1.0 version.

Reply:
Agreed.